### PR TITLE
Wait for checkpoint directory availability in JournalToolTest

### PR DESCRIPTION
**Pull request by bradyoo**

### CODE REVIEW

File: tests/src/test/java/alluxio/client/cli/JournalToolTest.java [hunk 1]
```diff
@@ -254,10 +256,27 @@ private long getCurrentCopyCatSnapshotIndex(String journalFolder) throws Throwab
     return currentSnapshotIdx.get();
   }
 
-  private String findCheckpointDir() throws IOException {
-    List<Path> checkpoint = Files.list(mDumpDir.toPath())
-        .filter(p -> p.toString().contains("checkpoints-")).collect(toList());
-    assertEquals("Unexpected checkpoint list: " + checkpoint, 1, checkpoint.size());
+  private String findCheckpointDir() throws Exception {
+    final List<Path> checkpoint = new ArrayList<>();
+    CommonUtils.waitFor("Checkpoint list:", () -> {
```
Comment (gpang) [inline tests/src/test/java/alluxio/client/cli/JournalToolTest.java:261]: Can we just do the "wait" part for getting the list of paths? Once we get the non-empty list of paths, we can just do the previous/existing assert of the size? The current implementation of this function is doing both the condition to terminate the wait, as well as asserts. I'd prefer this wait function only determines the termination condition for the wait, and do the assert outside of the wait.

**Comment by bradyoo** [comment]
Applied feedback.

**Comment by bradyoo** [comment]
alluxio-bot, test this please

File: tests/src/test/java/alluxio/client/cli/JournalToolTest.java [hunk 2]
```diff
@@ -254,10 +256,28 @@ private long getCurrentCopyCatSnapshotIndex(String journalFolder) throws Throwab
...
+      }
+
+      if (checkpointTest.size() == 0) {
+        return false;
+      }
+      checkpoint.addAll(checkpointTest);
+      return true;
+    }, WaitForOptions.defaults().setTimeoutMs(500));
+
+    if (checkpoint.size() > 1) {
```
Comment (gpang) [inline tests/src/test/java/alluxio/client/cli/JournalToolTest.java:277]: Can we do an `assertEquals(1, checkpoint.size());` instead? Or, we can just use the previous code assert line?

Comment (bradyoo) [reply tests/src/test/java/alluxio/client/cli/JournalToolTest.java:277]: Oh duh yeah you are right.

**Comment by bradyoo** [comment]
feedback applied though i expect no jenkins PR build is passing from the looks of it.

**Comment by bradyoo** [comment]
alluxio-bot, test this please.

**Comment by bradyoo** [comment]
Closing because this doesn't seem to fix the issue.

**Pull request closed without merging**